packages/contracts/test/forwarder/forwarder.ts

```diff
@@ -0,0 +1,306 @@
+            const takerZRXBalanceBeforeWithdraw = afterBuyBalances[takerAddress][zrxToken.address];
+            const takerForwarderBalance = await forwarderContract.balanceOf.callAsync(takerAddress);
+            const txHash = await forwarderContract.withdrawZRX.sendTransactionAsync(takerForwarderBalance, {
+                from: takerAddress,
+            });
+            const newBalances = await erc20Wrapper.getBalancesAsync();
+            const takerZRXBalanceAfterWithdraw = newBalances[takerAddress][zrxToken.address];
+            expect(takerZRXBalanceAfterWithdraw).to.be.bignumber.eq(
+                takerZRXBalanceBeforeWithdraw.plus(takerForwarderBalance),
+            );
+            const takerForwarderBalanceAfterWidthdraw = await forwarderContract.balanceOf.callAsync(takerAddress);
+            expect(takerForwarderBalanceAfterWidthdraw).to.be.bignumber.eq(new BigNumber(0));
+        });
+    });
+    describe('quote', () => {
+        it('buyTokensQuote - no fees', async () => {
+            const fillAmount = signedOrder.takerAssetAmount.div(4);
+            const takerBalanceBefore = erc20Balances[takerAddress][defaultMakerAssetAddress];
+            const sellQuote = await forwarderWrapper.buyTokensQuoteAsync([signedOrder], [], fillAmount);
+            const tx = await forwarderWrapper.buyTokensAsync([signedOrder], [], fillAmount, takerAddress);
+            const newBalances = await erc20Wrapper.getBalancesAsync();
+            const takerBalanceAfter = newBalances[takerAddress][defaultMakerAssetAddress];
+            expect(takerBalanceAfter).to.be.bignumber.eq(takerBalanceBefore.plus(sellQuote.makerAssetFilledAmount));
+        });
+        it('buyTokensQuote - fee abstraction', async () => {
+            const fillAmount = signedOrder.takerAssetAmount.div(2);
+            const takerBalanceBefore = erc20Balances[takerAddress][defaultMakerAssetAddress];
+            const buyTokensQuote = await forwarderWrapper.buyTokensQuoteAsync([orderWithFees], [feeOrder], fillAmount);
+            const tx = await forwarderWrapper.buyTokensAsync([orderWithFees], [feeOrder], fillAmount, takerAddress);
+            const newBalances = await erc20Wrapper.getBalancesAsync();
+            const takerBalanceAfter = newBalances[takerAddress][defaultMakerAssetAddress];
+            expect(takerBalanceAfter).to.be.bignumber.eq(
+                takerBalanceBefore.plus(buyTokensQuote.makerAssetFilledAmount),
+            );
+        });
+    });
+    describe('Non fungible tokens', async () => {
+        it('buys non fungible tokens', async () => {
+            const makerAssetId = erc721MakerAssetIds[0];
+            signedOrder = orderFactory.newSignedOrder({
+                makerAssetAmount: new BigNumber(1),
+                makerAssetData: assetProxyUtils.encodeERC721ProxyData(erc721Token.address, makerAssetId),
+            });
+            const tx = await forwarderWrapper.buyNFTTokensAsync([signedOrder], [feeOrder], takerAddress);
```

Comment:
Let's at least check the tx returned has status '1'. Is there nothing else worth checking? Like the actual balances post trade of the two participants perhaps.